medmodels/medrecord/medrecord.py

```diff
@@ -591,89 +601,130 @@ def add_nodes(
+        """Adds nodes to the MedRecord instance from one or more Pandas DataFrames. Optionally assigns them to a group.
 
         This method accepts either a single tuple or a list of tuples, where each tuple
-        consists of a Pandas DataFrame and an index column string.
+        consists of a Pandas DataFrame and an index column string. If a group is
+        specified, the nodes are added to the group.
 
         Args:
             nodes (Union[PandasNodeDataFrameInput, List[PandasNodeDataFrameInput]]):
                 A tuple or list of tuples, each with a DataFrame and index column.
+            group (Optional[Group]): The name of the group to add the nodes to. If not
+                specified, the nodes are added to the MedRecord without a group.
 
         Returns:
             None
         """
-        return self.add_nodes_polars(
+        self.add_nodes_polars(
             [process_nodes_dataframe(nodes_df) for nodes_df in nodes]
             if isinstance(nodes, list)
-            else [process_nodes_dataframe(nodes)]
+            else [process_nodes_dataframe(nodes)],
+            group,
         )
 
     def add_nodes_polars(
-        self, nodes: Union[PolarsNodeDataFrameInput, List[PolarsNodeDataFrameInput]]
+        self,
+        nodes: Union[PolarsNodeDataFrameInput, List[PolarsNodeDataFrameInput]],
+        group: Optional[Group] = None,
     ) -> None:
-        """Adds nodes to the MedRecord instance from one or more Polars DataFrames.
+        """Adds nodes to the MedRecord instance from one or more Polars DataFrames. Optionally assigns them to a group.
 
         This method accepts either a single tuple or a list of tuples, where each tuple
-        consists of a Polars DataFrame and an index column string.
+        consists of a Polars DataFrame and an index column string. If a group is
+        specified, the nodes are added to the group.
 
         Args:
             nodes (Union[PolarsNodeDataFrameInput, List[PolarsNodeDataFrameInput]]):
                 A tuple or list of tuples, each with a DataFrame and index column.
+            group (Optional[Group]): The name of the group to add the nodes to. If not
+                specified, the nodes are added to the MedRecord without a group.
 
         Returns:
             None
         """
-        return self._medrecord.add_nodes_dataframes(
+        self._medrecord.add_nodes_dataframes(
             nodes if isinstance(nodes, list) else [nodes]
         )
 
+        if group is not None:
+            if not self.contains_group(group):
+                self.add_group(group)
+
+            if isinstance(nodes, list):
+                nodes_indices = [
+                    nodes for node in nodes for nodes in node[0][node[1]].to_list()
+                ]
+            else:
+                nodes_indices = nodes[0][nodes[1]].to_list()
+
+            self.add_node_to_group(group, nodes_indices)
```

Comment:
```suggestion
        if group is None:
            return

        if not self.contains_group(group):
            self.add_group(group)

        if isinstance(nodes, list):
            node_indices = [
                nodes for node in nodes for nodes in node[0][node[1]].to_list()
            ]
        else:
            node_indices = nodes[0][nodes[1]].to_list()

        self.add_node_to_group(group, node_indices)
```

early return and renamed `nodes_indices` to `node_indices`